src/vtok_agent/src/imds.rs

```diff
@@ -124,21 +159,40 @@ impl ImdsCache {
     }
 
     fn fetch_role_arn() -> Result<String, Error> {
-        let profile_arn =
-            serde_json::from_str::<IamInfo>(&Self::fetch("/latest/meta-data/iam/info")?)
-                .and_then(|info| Ok(info.instance_profile_arn))
-                .map_err(Error::ParseError)?;
         let role_name = Self::fetch_role_name()?;
 
-        let role_arn = profile_arn
-            .split(":")
-            .map(|word| match word.find("instance-profile/") {
-                Some(_) => format!("role/{}", role_name),
-                None => word.to_string(),
-            })
-            .collect::<Vec<String>>()
-            .join(":");
-        Ok(role_arn)
+        let output = Command::new("aws")
+            .arg("iam")
+            .arg("get-role")
```

Comment:
I don't think it is a good idea to make this call here, this automatically implies that we need to update our documentation so that the role iam  attached policy allows calling of GetRole API, can't we just fix this logic, so that it takes into consideration that "/" is a valid character in a role name ?

Reply:
This is not about the "/" character, it's just that the current logic, that uses IMDS info, would not get the whole IAM role path as IMDS does not provide that. It provides the role name.

So if we have /path/to/role-name, IMDS would provide role-name, not the whole /path/to/role-name.

True, as mentioned in the commit message, the documentation would need to be updated to include the policy update to be able to get the IAM role info. If there is another option to get the role arn (for IAM roles both with and without paths), open to discuss.

Reply:
I disagree here:
You have the role name and you have the account, we should keep the building of the role arn locally without calling outside as we are currently doing and taking into consideration that a "/" might be part of the role name or the instance profile.

Reply:
The previous logic is still kept as a fallback, if the get-role call fails. It's just the path info is missing from IMDS, only the role-name is returned.